### ISSUE TITLE
Radiogroups "choices" array has different type.

### ISSUE THREAD

**Issue by nathan-at-jisc** [surveyjs/survey-creator#2060]
Sorry if this is too lengthy, so I'm going to start with an example.

Let's say I have a survey, with one question, and that's a `checkbox` question (although this "problem" persists on `radiogroup`, `matrix` etc). I can edit the checkbox to show different values, i.e

```json
{
    "pages": [
        {
            "name": "page1",
            "elements": [
                {
                    "type": "checkbox",
                    "name": "question1",
                    "choices": [
                        "item1",
                        "item2",
                        "item3"
                    ]
                }
            ]
        }
    ]
}
```

However, suppose I change the `text` of `item`, it now looks like:

```json
{
    "pages": [
        {
            "name": "page1",
            "elements": [
                {
                    "type": "checkbox",
                    "name": "question1",
                    "choices": [
                        {
                            "value": "item1",
                            "text": "Item 1"
                        },
                        "item2",
                        "item3"
                    ]
                }
            ]
        }
    ]
}
```

At Jisc we're using SurveyJS as the `creator` and `viewer`, but are creating our own analytics library: https://github.com/JiscSD/surveyjs-analytics-package. It's far from complete, but this is being used for CSV downloads, aggregates and individual responses and creates a *standard* template.

A lot of the issues we face with the code are around the possibility that `choices`, `rows`, `columns`, can be either a `tring` or an `object` with 2 properties. Would it be possible to remove the `string` and if there was no `choices.$.text`, it would always be:

```json
{
    "pages": [
        {
            "name": "page1",
            "elements": [
                {
                    "type": "checkbox",
                    "name": "question1",
                    "choices": [
                        {
                            "value": "item1",
                            "text": "item1"
                        },
                        {
                            "value": "item2",
                            "text": "item2"
                        },
                        {
                            "value": "item3",
                            "text": "item3"
                        },
                    ]
                }
            ]
        }
    ]
}
```

This way, we can remove uncertainty about the type of `choice`. In essence the solution here is to say, if there's no `text` set it equal to the `value`.

Would this be possible for SurveyJS to handle, or even have it as a paramater in the Survey Creator so that this happens for us?

**Comment by nathan-at-jisc** [comment]
Is there anyway we could do it so that if no `text` is present, we default `text` to equal `value`? If I tried to do a PR for that would that be OK?

**Comment by andrewtelnov** [comment]
@nathan-at-jisc I have implemented the functionality via this [PR](https://github.com/surveyjs/survey-library/pull/3899). It will be available in the next minor release, [v1.9.7](https://surveyjs.io/WhatsNew#v1.9.7).
You should add the following code:
`Survey.settings.itemValueAlwaysSerializeText = true;`

Thank you,
Andrew

**Comment by CarterQC** [comment]
Sorry to revive this, is there a setting for this on the creator side that will work on the creator side?

i.e. right now when I create a rating, it has no rateValues field by default.  Is there a way to always have choices/rate values/etc. be in this format: `{"text": "...", "value": ...}`
